pkgs/servers/sql/cockroachdb/default.nix

```diff
@@ -1,70 +1,52 @@
-{ lib, stdenv, buildGoPackage, fetchurl
-, cmake, xz, which, autoconf
-, ncurses6, libedit, libunwind
-, installShellFiles
-, removeReferencesTo, go
+{ lib
+, stdenv
+, buildFHSEnv
 }:
 
 let
-  darwinDeps = [ libunwind libedit ];
-  linuxDeps  = [ ncurses6 ];
-
-  buildInputs = if stdenv.isDarwin then darwinDeps else linuxDeps;
-  nativeBuildInputs = [ installShellFiles cmake xz which autoconf ];
-
-in
-buildGoPackage rec {
-  pname = "cockroach";
+  name = "cockroach";
   version = "20.1.8";
 
-  goPackagePath = "github.com/cockroachdb/cockroach";
-
-  src = fetchurl {
-    url = "https://binaries.cockroachdb.com/cockroach-v${version}.src.tgz";
-    sha256 = "0mm3hfr778c7djza8gr1clwa8wca4d3ldh9hlg80avw4x664y5zi";
-  };
-
-  env.NIX_CFLAGS_COMPILE = toString (lib.optionals stdenv.cc.isGNU [ "-Wno-error=deprecated-copy" "-Wno-error=redundant-move" "-Wno-error=pessimizing-move" ]);
-
-  inherit nativeBuildInputs buildInputs;
+  # For several reasons building cockroach from source has become
+  # nearly imposible. See https://github.com/NixOS/nixpkgs/pull/152626
+  # Therefore we use the pre-build release binary and wrap it with buildFHSUserEnv to
+  # work on nix.
+  # You can generate the hashes with
+  # nix store prefetch-file <url>
+  src = builtins.fetchurl (lib.getAttr stdenv.system {
```

Comment:
```suggestion
  src = fetchurl (lib.getAttr stdenv.system {
```

Reply:
I just noticed that `fetchurl` does not uncompress the tarball. I switched to `fetchzip`. Unfornutately this now results in another hash than `nix store pre-fetch-file` prints.

Reply:
`nix flake prefetch` seems to be the correct command.